### ISSUE TITLE
set-lookup-handlers! and minor mode

### ISSUE THREAD

**Issue by kvaneesh** [doomemacs/doomemacs#7905]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

The documentation states 
```

Consecutive `set-lookup-handlers!' calls will overwrite previously defined
handlers for MODES. If used on minor modes, they are stacked onto handlers
defined for other minor modes or the major mode it's activated in.
```

IIUC, when used with minor modes it should retain the existing lookup handlers and add the new handers to the list. 

An example like below 

     (set-lookup-handlers! 'ggtags-mode
       :definition #'ggtags-find-definition
       :references #'ggtags-find-reference)

Doesn't do that. After the above, enabling gtags-mode change the overwrite the lookup handlers. 

### Current behavior

Value of +lookup-references-functions
```

(+lookup-thesaurus-definition-backend-fn
 +lookup-xref-references-backend-fn
 +lookup-project-search-backend-fn)
```

after enabling minor mode ggtags-mode

```
(ggtags-find-reference
 t)
```

### Steps to reproduce

Evaluate set-lookup-handlers! with a minor mode and check for +lookup-references-functions

### System Information

Can recreate with any config

**Comment by hlissner** [comment]
It does, but the problem is that the lookup system stops on the first handler to return non-nil OR change the window configuration (i.e. open a new window/buffer). You'll need to inspect `ggtags-find-definition` et co to see what they return or what it does. If it's asynchronous though, things get real complicated; you'll either have to do some sort of manual blocking, or return `'deferred` *or* append those ggtags functions to `+lookup-references-functions` manually. It's messy, because there's no standardization around these things.

In any case, I'll close this since there is no fixable issue here, but feel free to follow up here.

**Comment by kvaneesh** [comment]
Should we then update the documentation 
```

Consecutive `set-lookup-handlers!' calls will overwrite previously defined
handlers for MODES. If used on minor modes, they are stacked onto handlers
defined for other minor modes or the major mode it's activated in.
```
and avoid mentioning they get stacked?

**Comment by hlissner** [comment]
No, they *are* stacked, the same way functions are stacked in any conventional Emacs hook variable. E.g. In the example above, these are the global lookup functions:

```
(+lookup-thesaurus-definition-backend-fn
 +lookup-xref-references-backend-fn
 +lookup-project-search-backend-fn)
```

And these are the buffer local ones:

```
(ggtags-find-reference
 t)
```

That `t` means "fall back to the global ones here".
